common/src/main/java/org/apache/sedona/common/geography/Constructors.java

```diff
@@ -67,4 +68,174 @@ public static Geography geogCollFromText(String wkt, int srid) throws ParseExcep
+  }
+
+  public static Geometry geogToGeometry(
+      Geography geography, GeometryFactory geometryFactory, int srid) {
+    if (geography == null) return null;
+
+    Geography.GeographyKind kind = Geography.GeographyKind.fromKind(geography.getKind());
+    switch (kind) {
+      case SINGLEPOINT:
+      case POINT:
+        return pointToGeom(geography, geometryFactory);
+      case SINGLEPOLYLINE:
+      case POLYLINE:
+        return polylineToGeom(geography, geometryFactory);
+      case POLYGON:
+      case MULTIPOLYGON:
+        return polygonToGeom(geography, geometryFactory);
+      case GEOGRAPHY_COLLECTION:
+        return collectionToGeom(geography, geometryFactory);
+      default:
+        return null;
+    }
+  }
+
+  // POINT/SINGLEPOINT
+  private static Geometry pointToGeom(Geography g, GeometryFactory gf) {
+    if (g instanceof SinglePointGeography) {
+      S2Point p = ((SinglePointGeography) g).getPoints().get(0);
+      S2LatLng ll = S2LatLng.fromPoint(p);
+      return gf.createPoint(new Coordinate(ll.lngDegrees(), ll.latDegrees()));
+    } else if (g instanceof PointGeography) {
+      List<S2Point> pts = ((PointGeography) g).getPoints();
+      Coordinate[] cs = new Coordinate[pts.size()];
+      for (int i = 0; i < pts.size(); i++) {
+        S2LatLng ll = S2LatLng.fromPoint(pts.get(i));
+        cs[i] = new Coordinate(ll.lngDegrees(), ll.latDegrees());
+      }
+      return gf.createMultiPointFromCoords(cs);
+    }
+    return null;
+  }
+
+  // POLYLINE/SINGLEPOLYLINE
+  private static Geometry polylineToGeom(Geography g, GeometryFactory gf) {
+    if (g instanceof PolylineGeography) {
+      List<S2Polyline> lines = ((PolylineGeography) g).getPolylines();
+      LineString[] lss = new LineString[lines.size()];
+      for (int i = 0; i < lines.size(); i++) {
+        S2Polyline pl = lines.get(i);
+        int n = pl.numVertices();
+        Coordinate[] cs = new Coordinate[n];
+        for (int k = 0; k < n; k++) {
+          S2LatLng ll = S2LatLng.fromPoint(pl.vertex(k));
+          cs[k] = new Coordinate(ll.lngDegrees(), ll.latDegrees());
+        }
+        lss[i] = gf.createLineString(cs);
+      }
+      return lss.length == 1 ? lss[0] : gf.createMultiLineString(lss);
+    }
+    return null;
+  }
+
+  // POLYGON / MULTIPOLYGON
+  private static Geometry polygonToGeom(Geography g, GeometryFactory gf) {
+    if (g instanceof PolygonGeography) {
+      S2Polygon s2p = ((PolygonGeography) g).polygon;
+      return s2LoopsToJts(s2p.getLoops(), gf);
+    } else if (g instanceof MultiPolygonGeography) {
+      List<Geography> parts = ((MultiPolygonGeography) g).getFeatures();
+      Polygon[] polys = new Polygon[parts.size()];
+      for (int i = 0; i < parts.size(); i++) {
+        polys[i] = (Polygon) s2LoopsToJts(((PolygonGeography) parts.get(i)).polygon.getLoops(), gf);
+      }
+      return polys.length == 1 ? polys[0] : gf.createMultiPolygon(polys);
```

Comment:
I think it is OK to always return multi polygon to preserve its original geography type.